### DISCUSSION TITLE
Customize for Dynamic Error Message

### DISCUSSION THREAD

**Discussion by Louis116** [react-hook-form/react-hook-form#9801] [Q&A]
Greetings,

I would like to unify the error message with a function for different error case, 
Instead of returning a static string , I would like to show the validation value that have been set into that specific field for example: max / min length,
but could not find a way to reach the rules value I have set for,

Would there be any possible way to retreive the value I have set and show back in the error message?

example as below:

```
const getFieldErrorMessage = (fieldError?: FieldError) => {
    let errormessage = ''
    if (fieldError) {
        const fieldName = fieldError.ref?.name ?? ''
        switch (fieldError.type) {
            case 'minLength':
                errormessage = 'Minimum words not met' // want to show as "Minimum words of ${set validation rules value} not met" instead
                break
            default:
                errormessage = `Field ${fieldName} error: ${fieldError.type}`
                break
        }
    }
    return errormessage
}
 <input {...register('selfIntro', {
                                    required: true,
                                    minLength: 10,
                                })}
                            />
<span>{getFieldErrorMessage(errors.selfIntro)}</span>
```
and error should throw something like: Minimum words of 10 not met.

Big thanks for any suggestion!

**Comment by Moshyfawn** [comment]
Would doing this help you:
```js
{
  minLength: {
    value: 10,
    message: "Please, enter at least 10 characters"
  }
}
```

You can get the error like
```js
errors.field?.message
```

**Reply by Louis116** [reply]
Thank you for your suggestion,
and I just notice that first line of code from my example is missing
`const getFieldErrorMessage = (fieldError?: FieldError) => {`

What I am trying to archive is by one getFieldErrorMessage function, i could get the custumized preset errror message.

so if there are hundreds or thousands of input field that have different min length rules, i would not need to repeat the message one by one, by one function, it could return its corrsponding error and the reason.

the main question would be is there a way to retrive the min/max lenght value I have set in the rule and display back on the message.

hence for your suggest of static error message is not the solution i am looking for.

anyway, thanks a again

I would left an other example for easier understanding:
```
 <input {...register('selfIntro', {
                                    required: true,
                                    minLength: 10,
                                })}
                            />
<span>{getFieldErrorMessage(errors.selfIntro)}</span> // this should return: Minimum words of 10 not met"
 <input {...register('name', {
                                    required: true,
                                    minLength: 5,
                                })}
                            />
<span>{getFieldErrorMessage(errors.name)}</span> // this should return: Minimum words of 5 not met"
